File: insertion_sort/insertion_sort.go
package insertion_sort

import "fmt"

func InsertionSort() {
	array := []int{10, 9, 8, 27, 16, 15, 4, 3, 3, 3, 2, 1}
	result := insertionSort(array)
	fmt.Println("Sorted array is:", result)
}

func insertionSort(array []int) []int {
	for i := 1; i < len(array); i++ {
		for j := i; j > 0 && array[j] < array[j-1]; j-- {
			array[j], array[j-1] = array[j-1], array[j]
		}
	}
	return array
}

File: insertion_sort/insertion_sort.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> insertionSort(vector<int> v) {
    for (int i = 1; i < v.size(); i++) {
        for (int j = i; j > 0 && v[j] < v[j - 1]; j--) {
            int temp = v[j];
            v[j] = v[j - 1];
            v[j - 1] = temp;
        }
    }
    return v;
}
